internal/error.go

```diff
@@ -30,11 +30,16 @@ type ecsError struct {
 }
 
 func NewError(err error) zapcore.ObjectMarshaler {
+	if e, ok := err.(zapcore.ObjectMarshaler); ok {
+		return e
+	}
```

Comment:
@dmexe why would you even call `NewError` if already holding a `zapcore.ObjectMarshaler`?